### PR TITLE
chore: add CODEOWNERS

### PR DESCRIPTION
**Pull request by dalelane**
Based on initial list in
https://github.com/asyncapi/community/discussions/189

Signed-off-by: Dale Lane <dale.lane@uk.ibm.com>

### CODE REVIEW

**Comment by derberg** [comment]
@AGurlhosur @dan-r @KieranM1999 @lewis-relph @JEFFLUFC folks please approve this PR, so you officially indicate you want to be CODEOWNERS here, and you understand that you can also join the AsyncAPI's Technical Steering Committee after this PR is merged (I suggest one PR updates [this file](https://github.com/asyncapi/community/blob/master/TSC_MEMBERS.json) and not everyone separately)

@dalelane please remove me and @fmvilas from the list. We did nothing here, it is all your glory folks, you do not need us here. Also please remember that everyone in AsyncAPI, maintainers, and contributors work through forks, no exceptions, so even maintainers should open PR from forks, no custom branches in the upstream. [Here](https://github.com/asyncapi/.github/blob/master/git-workflow.md) is a nice instruction on how to properly configure fork, to not notice a difference when working with upstream.

**Review by dan-r — APPROVED** [review]
Looks good to me and happy to be listed as a codeowner.

**Review by lewis-relph — APPROVED** [review]

**Review by KieranM1999 — APPROVED** [review]

**Review by JEFFLUFC — APPROVED** [review]

**Review by AGurlhosur — APPROVED** [review]

**Pull request merged**
